File: ios/card-detector-tflite/cardDetectorTflite/cardDetectorTflite.m
#import <VisionCamera/FrameProcessorPlugin.h>
#import <VisionCamera/FrameProcessorPluginRegistry.h>

#if __has_include(""card-detector-tflite"/"card-detector-tflite"-Swift.h")
#import ""card-detector-tflite"/"card-detector-tflite"-Swift.h"
#else
#import ""card-detector-tflite"-Swift.h"
#endif

VISION_EXPORT_SWIFT_FRAME_PROCESSOR(cardDetectorTflitePlugin, detectCard)

File: ios/card-detector-tflite/cardDetectorTflite/cardDetectorTflite.swift
import VisionCamera

@objc(cardDetectorTflitePlugin)
public class cardDetectorTflitePlugin: FrameProcessorPlugin {
  public override init(options: [AnyHashable: Any]! = [:]) {
    super.init(options: options)
  }

  public override func callback(_ frame: Frame, withArguments arguments: [AnyHashable: Any]?) -> Any? {
    let buffer = frame.buffer
    let orientation = frame.orientation
    // code goes here

    guard let modelPath = Bundle.main.path(forResource: "mobilenet_quant_v1_224", ofType: "tflite") else { print("Failed to load the model."); return nil }

    var options = InterpreterOptions()
    options.threadCount = 1
    do {
      // Initialize as Interpreter
      interpreter = try Interpreter(modelPath: modelPath, options: options)
      // Allocate memory for the input tensor
      try interpreter.allocateTensors()
    } catch let error {
      print("Failed to create the interpreter with error: \(error.localizedDescription)")
      return nil
    }
    return nil
  }
}